article/app/views/package.scala

```diff
@@ -36,6 +38,10 @@ object MainCleaner {
 object BodyCleaner {
   def apply(article: Article, html: String, amp: Boolean)(implicit request: RequestHeader) = {
     implicit val edition = Edition(request)
+
+    val shouldShowAds = !article.content.shouldHideAdverts && article.metadata.section != "childrens-books-site"
```

Comment:
We have [`commercial-features.js`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/commercial/commercial-features.js) which does codify some of these checks (see `var sensitiveContent`), which you could plug in.